homeassistant/components/dormakaba_dkey/config_flow.py

```diff
@@ -0,0 +1,156 @@
+            return self.async_abort(reason="no_devices_found")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_ADDRESS): vol.In(
+                    {
+                        service_info.address: (
+                            f"{service_info.name} ({service_info.address})"
+                        )
+                        for service_info in self._discovered_devices.values()
+                    }
+                ),
+            }
+        )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=data_schema,
+            errors=errors,
+        )
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> FlowResult:
+        """Handle the Bluetooth discovery step."""
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+        self._discovery_info = discovery_info
+        name = self._discovery_info.name or self._discovery_info.address
+        self.context["title_placeholders"] = {"name": name}
+        return await self.async_step_bluetooth_confirm()
+
+    async def async_step_bluetooth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle bluetooth confirm step."""
+        # mypy is not aware that we can't get here without having these set already
+        assert self._discovery_info is not None
+
+        if user_input is None:
+            name = self._discovery_info.name or self._discovery_info.address
+            return self.async_show_form(
+                step_id="bluetooth_confirm",
+                description_placeholders={"name": name},
+            )
+
+        return await self.async_step_associate()
+
+    async def async_step_associate(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle associate step."""
+        # mypy is not aware that we can't get here without having these set already
+        assert self._discovery_info is not None
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="associate", data_schema=STEP_ASSOCIATE_SCHEMA
+            )
+
+        errors = {}
+        if not self._lock:
+            self._lock = DKEYLock(self._discovery_info.device)
+        lock = self._lock
+
+        try:
+            association_data = await lock.associate(user_input["activation_code"])
+        except BleakError:
+            return self.async_abort(reason="cannot_connect")
+        except dkey_errors.InvalidActivationCode:
+            errors["base"] = "invalid_code"
+        except dkey_errors.WrongActivationCode:
+            errors["base"] = "wrong_code"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            return self.async_abort(reason="unknown")
+        else:
+            self._abort_if_unique_id_configured()
```

Comment:
I'm not sure if this is wanted, or if we should just proceed and replace the config entry instead?

Reply:
When could this abort?

Reply:
It could indeed never happen, removed now 👍